src/views/Lottery/components/AllHistoryCard/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useRef, useEffect } from 'react'
+import styled from 'styled-components'
+import { Card, Text, Skeleton, CardHeader, Box } from '@pancakeswap/uikit'
+import { useTranslation } from 'contexts/Localization'
+import { useLottery } from 'state/lottery/hooks'
+import { fetchLottery } from 'state/lottery/helpers'
+import { LotteryStatus } from 'config/constants/types'
+import RoundSwitcher from './RoundSwitcher'
+import { getDrawnDate, processLotteryResponse } from '../../helpers'
+import PreviousRoundCardBody from '../PreviousRoundCard/Body'
+import PreviousRoundCardFooter from '../PreviousRoundCard/Footer'
+
+const StyledCard = styled(Card)`
+  width: 100%;
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    width: 756px;
+  }
+`
+
+const StyledCardHeader = styled(CardHeader)`
+  z-index: 2;
+  background: none;
+  border-bottom: 1px ${({ theme }) => theme.colors.cardBorder} solid;
+`
+
+const YourHistoryCard = () => {
+  const { t } = useTranslation()
+  const {
+    currentLotteryId,
+    currentRound: { status, isLoading },
+  } = useLottery()
+  const currentLotteryIdAsInt = parseInt(currentLotteryId)
+  const mostRecentFinishedRoundId =
+    status === LotteryStatus.CLAIMABLE ? currentLotteryIdAsInt : currentLotteryIdAsInt - 1
+  const [selectedRoundId, setSelectedRoundId] = useState(mostRecentFinishedRoundId.toString())
+  const [selectedLotteryInfo, setSelectedLotteryInfo] = useState(null)
+  const timer = useRef(null)
+
+  useEffect(() => {
+    setSelectedLotteryInfo(null)
+
+    const fetchLotteryData = async () => {
+      const lotteryData = await fetchLottery(selectedRoundId)
+      const processedLotteryData = processLotteryResponse(lotteryData)
+      setSelectedLotteryInfo(processedLotteryData)
+    }
+
+    timer.current = setInterval(() => {
+      if (selectedRoundId) {
+        fetchLotteryData()
+      }
+      clearInterval(timer.current)
+    }, 1000)
+
+    return () => clearInterval(timer.current)
+  }, [selectedRoundId])
```

Comment:
Each time the selected round Id changes - clear the current lottery data to trigger a loading state, and start a 1s timer before data is fetched (to prevent unnecessary extra fetches as the user inputs numbers)